File: perf/bench/matmul/matmul.go
package main

import "fmt"

var a, b, c [200][200]int64

func main() {
	for i := 0; i < 200; i++ {
		for j := 0; j < 200; j++ {
			a[i][j] = int64((i + j) % 7)
			b[i][j] = int64((i * j) % 5)
			c[i][j] = 0
		}
	}
	for rep := 0; rep < 20; rep++ {
	for i := 0; i < 200; i++ {
		for k := 0; k < 200; k++ {
			aik := a[i][k]
			for j := 0; j < 200; j++ {
				c[i][j] += aik * b[k][j]
			}
		}
	}
	}
	var sum int64 = 0
	for i := 0; i < 200; i++ {
		sum += c[i][i]
	}
	fmt.Println(sum)
}

File: perf/bench/matmul/matmul.zig
const std = @import("std");
var a: [200][200]i64 = undefined;
var b: [200][200]i64 = undefined;
var c: [200][200]i64 = undefined;
pub fn main() !void {
    for (0..200) |i| for (0..200) |j| {
        a[i][j] = @intCast((i + j) % 7);
        b[i][j] = @intCast((i * j) % 5);
        c[i][j] = 0;
    };
    for (0..20) |_| for (0..200) |i| for (0..200) |k| {
        const aik = a[i][k];
        for (0..200) |j| c[i][j] += aik * b[k][j];
    };
    var sum: i64 = 0;
    for (0..200) |i| sum += c[i][i];
    var _pbuf: [64]u8 = undefined;
    const _ps = try std.fmt.bufPrint(&_pbuf, "{d}\n", .{sum});
    _ = try std.posix.write(1, _ps);
}
